components/core/dl.conf

```diff
@@ -1 +1 @@
-{"max_downloads": 2, "down_folder": "download/file/path", "size_limit":100000, "max_age":1, "min_rating":3, "max_bandwidth": 300000, "aria_server": "ws://localhost:6800/jsonrpc", "tmp_folder": "/path/to/tmp/folder"}
+{"max_downloads": 2, "down_folder": "../../downloads", "size_limit":100000, "max_age":1, "min_rating":3, "max_bandwidth": 300000, "aria_server": "ws://aria2c:6800/jsonrpc", "tmp_folder": "/path/to/tmp/folder"}
```

Comment:
Same config would be used by normal server run, but it will run in docker compose only as it use "aria2c" as host name.

Reply:
Also, change the tmp_foler path to "/tmp/bassa/"

Reply:
@vivonk I think we need to use **aria2c** as **hostname** because **dl.conf** resides in the Bassa API server. Moreover, I am not exposing ports for **aria2c** to host so accessing it on localhost is not possible, right? 
@vivonk Coming to download folder I have updated the path to the `downloads volume` which resides in the Bassa API container. Moreover downloads volume is mapped to the local downloads folder for Bassa API server to access.

Reply:
> Also, change the tmp_folder path to "/tmp/bassa/"

Sure sir 😄

Reply:
> @vivonk I think we need to use **aria2c** as **hostname** because **dl.conf** resides in the Bassa API server. Moreover, I am not exposing ports for **aria2c** to host so accessing it on localhost is not possible, right?
> @vivonk Coming to download folder I have updated the path to the `downloads volume` which resides in the Bassa API container. Moreover downloads volume is mapped to the local downloads folder for Bassa API server to access.

@kmehant this config will work fine in network behind seen of docker, but if we run the Bassa through simple scripts then "ws://aria2c:6800" will not be understood by connectors as "aria2c:6800" will not get resolved.

Reply:
I guess in solution of this, we can put a flag which can identify that which configurations have to be picked up for normal run and docker run. @agentmilindu @kmehant what do you say?

Reply:
@vivonk how about manually changing the configuration as **hostname** the only concerning thing to be changed which is the case with `DBCon.py` too.
Shall we have a guide on _How to get Bassa ready for development with/without docker_ mentioning the small changes which devs have to follow or automation script which does changes to files such as `dl.conf` and `DBCon.py` automatically getting dev ready? 
@vivonk @agentmilindu Thoughts?

Reply:
Let's say for this initial release, we can do that but after this we will automate this part.

Reply:
Sure :)

> Let's say for this initial release, we can do but after that we will automate this part.